pkg/operator/certrotationcontroller/satokensigner_controller.go

```diff
@@ -274,6 +285,33 @@ func (c *SATokenSignerController) syncWorker() error {
 	return nil
 }
 
+// isValidKeyPair checks if a public/private keys match i.e if the tokens signed
+// with private key can be validated using public key. if "pubKeyData" or "privKeyData"
+// contains multiple keys, it tries to validate the first available keys in byte slices
+func isValidKeyPair(pubKeyData, privKeyData []byte) error {
```

Comment:
you need to adjust the function name as well: `isValid*` implies returning a boolean, the convention when returning just an error would be `validate*`

Reply:
Done